apps/bluetooth/js/transfer.js

```diff
@@ -208,27 +206,46 @@ navigator.mozL10n.once(function showPanel() {
         var getRequest = storage.get(filepath);
 
         getRequest.onsuccess = function() {
-          defaultAdapter.sendFile(targetDevice.address, getRequest.result);
-          var msg = 'getFile succeed & file is sending...';
-          debug(msg);
-          if (--numberOfTasks === 0) {
-            transferred();
-          }
+          msg = 'getFile succeed & file is sending...';
+          sendFile(targetDevice.address, getRequest.result, msg);
         };
 
         getRequest.onerror = function() {
-          defaultAdapter.sendFile(targetDevice.address, blob);
-          var msg = 'getFile failed so that blob is sending without filename ' +
-                    getRequest.error && getRequest.error.name;
-          debug(msg);
-          if (--numberOfTasks === 0) {
-            transferred();
-          }
+          msg = 'getFile failed so that blob is sending without filename ' +
+                getRequest.error && getRequest.error.name;
+          sendFile(targetDevice.address, blob, msg);
         };
+      } else {
+        // The blob does not have name and filepath,
+        // pass the blob without name to send file directly.
+        msg = 'no filepath to get from device storage ' +
+              'so that blob is sending without filename ';
+        sendFile(targetDevice.address, blob, msg);
       }
     });
   }
 
+  /**
+   * Given address, blob(file), message to send file via platform.
+   *
+   * @access private
+   * @memberOf transfer
+   * @param {String} address - address of target device
+   * @param {Object} blob - blob(file) to send
+   * @param {String} msg - message for debugging only
+   */
+  function sendFile(address, blob, msg) {
+    defaultAdapter.sendFile(address, blob);
+    debug(msg);
+    checkTasksComplete();
+  }
```

Comment:
I like having this as a separate function, but think you should move the debug out of it.